src/Diagnostics.DataProviders/AscClient.cs

```diff
@@ -0,0 +1,378 @@
+                            }
+
+                            return value;
+                        }
+                    }
+                    else
+                    {
+                        string responseLogBody = await GetRequestDetailsForLogging(response);
+                        throw new Exception(string.Format("HTTP POST request to ASC failed. AppLens request Id : {0} ==> Details : {1}", this.requestId, responseLogBody));
+                    }
+                }
+            }
+            catch (TaskCanceledException ex)
+            {
+                if (cancellationToken != default(CancellationToken))
+                {
+                    throw new Exception(string.Format("The HTTP POST request to ASC with body {0} was cancelled as per the supplied cancellation token. AppLens request Id : {1}", jsonPostBody, requestId), ex);
+                }
+                else
+                {
+                    throw;
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<T> MakeHttpGetRequest<T>(string queryString,string apiUri, string apiVersion, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string requestUri = string.Empty;
+
+            if (string.IsNullOrEmpty(apiUri))
+            {
+                apiUri = this.apiUri;
+            }
+
+            if (string.IsNullOrEmpty(apiVersion))
+            {
+                apiVersion = this.apiVersion;
+            }
+
+            string authToken = await AscTokenService.Instance.GetAuthorizationTokenAsync();
+            try
+            {
+                if (string.IsNullOrEmpty(queryString))
+                {
+                    requestUri = baseUri + apiUri + "?api-version=" + apiVersion.Replace("?", string.Empty).Replace("api-version=", string.Empty);
+                }
+                else
+                {
+                    requestUri = baseUri + apiUri + "?" + queryString.Replace("?", string.Empty) + "&api-version=" + apiVersion.Replace("?", string.Empty).Replace("api-version=", string.Empty);
+                }
+
+                httpClient.DefaultRequestHeaders.UserAgent.Clear();
+                httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(new ProductHeaderValue(userAgent)));
+
+                using (HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Get, requestUri))
+                {
+                    requestMessage.Headers.Add("Authorization", authToken);
+
+                    httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", authToken);
+
+                    var response = await httpClient.SendAsync(requestMessage, cancellationToken);
+                    string responseContent = await response.Content.ReadAsStringAsync();
+                    if (response.IsSuccessStatusCode)
```

Comment:
The block in `if (response.IsSuccess..)` is the same as the block in the method above - can you refactor to a single method to avoid code copy?

Reply:
You can see similarity to code I added recently in `SupportObserverDataProviderBase.SendObserverRequestAsync`

Reply:
Good suggestion, followed the same practice. Helps me log stuff too.